tests/test_base_models.py

```diff
@@ -154,6 +154,71 @@ def next_state(self, x, u, dt):
         (times, inputs, states, outputs, event_states) = m.simulate_to_threshold(load, {'o1': 0.8}, **{'dt': 0.5, 'save_freq': 1.0})
         self.assertAlmostEqual(times[-1], 5.0, 5)
 
+    def test_size(self):
+        m = MockProgModel()
+        size = sys.getsizeof(m)
+        self.assertLess(size, 7500)
+
+        # Adding a parameter
+        m.parameters['test'] = 8675309
+        size2 = sys.getsizeof(m)
+
+        # Check that size increases
+        self.assertLess(size, size2)
+
+        # Size difference should be slightly more than the size of key & value
+        # The difference is overhead for the dict
+        # Note- have to use an uncommon number for this to work
+        # This is because of python's memory allocation
+        diff = size2 - size
+        sum_of_parts = sys.getsizeof('test') + sys.getsizeof(8675309)
+        self.assertLess(sum_of_parts, diff)
+        self.assertLess(diff-sum_of_parts, 100)
+
+        # Adding other attributes
+        m._test_value = 123456789
```

Comment:
'_test_value', is this a typo where it should be 'test_value'?

Reply:
Since we are adding an additional test, maybe the key should be called a bit more descriptive and state "additional_test_value", or something along those lines.

Reply:
I think its to let people know that the variable is meant to be private. 
https://www.geeksforgeeks.org/private-variables-python/

Reply:
Oh, good point.

I guess I just wrongly assumed it was weird that we were declaring a private variable after declaration and later passing a value into it that I didn't think it was a private variable.

Reply:
You're right that in practice it's strange to declare a private variable from outside of the scope of the class after its creation. But, I think it's ok for the sake of this test where the variable name doesn't matter